tests/sentry/sentry_metrics/test_gen_metrics_multiprocess_steps.py

```diff
@@ -0,0 +1,618 @@
+                Partition(Topic("topic"), 0),
+                1,
+                BROKER_TIMESTAMP,
+            )
+        ),
+    ]
+    # the outer message uses the last message's partition, offset, and timestamp
+    last = message_batch[-1]
+
+    outer_message = Message(Value(message_batch, last.committable))
+
+    with caplog.at_level(logging.ERROR):
+        new_batch = MESSAGE_PROCESSOR.process_messages(outer_message=outer_message)
+
+    # we expect just the valid counter_payload msg to be left
+    expected_msg = message_batch[0]
+    expected_new_batch = [
+        Message(
+            Value(
+                KafkaPayload(
+                    None,
+                    json.dumps(__translated_payload(counter_payloads[0])).encode("utf-8"),
+                    [("metric_type", b"c")],
+                ),
+                expected_msg.committable,
+            )
+        )
+    ]
+    compare_message_batches_ignoring_metadata(new_batch, expected_new_batch)
+    assert error_text in caplog.text
+
+
+@pytest.mark.django_db
+def test_process_messages_rate_limited(caplog, settings) -> None:
+    """
+    Test handling of `None`-values coming from the indexer service, which
+    happens when postgres writes are being rate-limited.
+    """
+    settings.SENTRY_METRICS_INDEXER_DEBUG_LOG_SAMPLE_RATE = 1.0
+    rate_limited_payload = deepcopy(distribution_payloads[0])
+    rate_limited_payload["tags"]["rate_limited_test"] = "true"
+
+    message_batch = [
+        Message(
+            BrokerValue(
+                KafkaPayload(None, json.dumps(counter_payloads[0]).encode("utf-8"), []),
+                Partition(Topic("topic"), 0),
+                0,
+                BROKER_TIMESTAMP,
+            )
+        ),
+        Message(
+            BrokerValue(
+                KafkaPayload(None, json.dumps(rate_limited_payload).encode("utf-8"), []),
+                Partition(Topic("topic"), 0),
+                1,
+                BROKER_TIMESTAMP,
+            )
+        ),
+    ]
+    # the outer message uses the last message's partition, offset, and timestamp
+    last = message_batch[-1]
+    outer_message = Message(Value(message_batch, last.committable))
+
+    message_processor = MessageProcessor(
+        get_ingest_config(UseCaseKey.PERFORMANCE, IndexerStorage.MOCK)
+    )
+    # Insert a None-value into the mock-indexer to simulate a rate-limit.
+    mock_indexer: Any = message_processor._indexer
+    rgx: Any = re.compile("^([c|s|d|g|e]):([a-zA-Z0-9_]+)/.*$").match(
+        distribution_payloads[0]["name"]
+    )
+    mock_indexer.indexer._strings[UseCaseID(rgx.group(2))][1]["rate_limited_test"] = None
+
+    with caplog.at_level(logging.ERROR):
+        new_batch = message_processor.process_messages(outer_message=outer_message)
+
+    # we expect just the counter_payload msg to be left, as that one didn't
+    # cause/depend on string writes that have been rate limited
+    expected_msg: Any = message_batch[0]
```

Comment:
this shouldn't need an annotation -- `message_batch` should be `list[Message]` and indexing it should give `Message` which should work fine ?

Reply:
It doesn't actually work, I managed pass typecheck now with
```python
expected_msg = message_batch[0]
assert isinstance(expected_msg.value, BrokerValue)
```